src/crda.ts

```diff
@@ -115,15 +116,18 @@ namespace Crda {
             const exitCode = await ghExec.exec(executable, args, finalExecOptions);
             ghCore.debug(`Exit code ${exitCode}`);
 
-            if (execOptions.ignoreReturnCode !== true && exitCode !== 0
-                // allow crda to exit with 2 indicating scan failure
-                && (executable !== CRDA_EXECUTABLE && exitCode !== 2)
+            const failNonCrda = executable !== CRDA_EXECUTABLE && exitCode !== 0;
+
+            // avoiding failure if exit code is 2 as if vulnerability is found exit code is 2
+            const failCrda = executable === CRDA_EXECUTABLE && exitCode !== 2 && exitCode !== 0;
+
+            if (execOptions.ignoreReturnCode !== true && (failCrda || failNonCrda)
```

Comment:
this is fairly hard to parse

Reply:
```javascript

let failed;
if (executable === CRDA_EXECUTABLE) {
	// crda exit 2 indicates a vulnerability was found, so that's an expected error
	failed = exitCode !== 0 && exitCode !== 2;
}
else {
	failed = exitCode !== 0;
}

if (failed && !execOptions.ignoreReturnCode) {
  // throw error etc
}
```

Reply:
This is much better, Thanks!